### PR TITLE
VI Arc Enabled Tutorial On Vanilla K8s Cluster ( Kubeadm- Single Node)

### PR DESCRIPTION
**Pull request by tshaiman**
- create a new tutorial for non-AKS based cluster
- updating the deployment script to fit kubeadm 
-

### CODE REVIEW

File: VideoIndexerEnabledByArc/aks/vi-edge-deployment-script.sh [hunk 1]
```diff
@@ -179,78 +179,81 @@ fi
...
-      if [[ "$connectedResult" == *"ERROR"* ]]; then
-        echo "AKS-Arc-connected connectivity Sanity test failed. Exiting"
-        exit 1
-      fi
-      echo -e "\tconnecting AKS to ARC-AKS -- ***done***"
+  #=============================================#
+  #======== Create Connected-ARC Cluster =======#
+  #=============================================#
+  echo -e "\tConnecting AKS to ARC-AKS -- ***start***"
+  az group create -g ${rg} --location ${region} --tags $tags
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/aks/vi-edge-deployment-script.sh:248]: Isn't `az create group` redundant here? The aks cluster was created in this RG before

File: VideoIndexerEnabledByArc/aks/vi-edge-deployment-script.sh [hunk 2]
```diff
@@ -295,6 +298,7 @@ if [[ $install_extension == "true" ]]; then
     echo -e "\tUpdating VI Extension - ***done***"
   else  
     echo -e "\tCreate New VI Extension - ***start***"
+    az group create -g ${rg} --location ${region}
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/aks/vi-edge-deployment-script.sh:301]: same

File: VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  echo "https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli"
+  az extension add --name connectedk8s
+  az extension add --name k8s-extension
+  az extension add --name aks-preview
+  az provider register --namespace Microsoft.Kubernetes
+  az provider register --namespace Microsoft.KubernetesConfiguration
+  az provider register --namespace Microsoft.ExtendedLocation
+
+  echo "install jq"
+  sudo apt-get update && apt-get install jq -y
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh:22]: This assumes the customer runs it in ubuntu. We should document it in the README

File: VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    --nodepool-name system \
+    --os-sku AzureLinux \
+    --node-count 2 \
+    --tier standard \
+    --generate-ssh-keys \
+    --network-plugin kubenet \
+    --tags $tags \
+    --node-resource-group $nodePoolRg \
+    --node-vm-size $nodeVmSize \
+    --node-os-upgrade-channel NodeImage --auto-upgrade-channel node-image
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh:133]: we can also enable image cleaner for the customer

File: VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+echo -e  "\tConnecting to AKS and getting credentials  - ***start***"
+az aks get-credentials --resource-group $rg --name $aks --admin --overwrite-existing
+echo "AKS connectivity Sanity test"
+kubectl get nodes
+echo -e "\tconnect aks cluster - ***done***"
+#=============================================#
+#============== Add ingress controller =======#
+#=============================================#
+echo -e "\tAdding ingress controller -- ***start***"
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.9.4/deploy/static/provider/cloud/deploy.yaml
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh:161]: I think 1.9.5 is the latest

File: VideoIndexerEnabledByArc/scripts/vi_extension_install.sh [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+version="1.0.41"                    #VI Extension version
+singleNodeStorageClass="local-path" #default storage class for single node cluster
+singleNodeAccessMode="ReadWriteOnce" #default access mode for single node cluster
+region="eastus"
+extensionName="vi-arc"
+namespace="video-indexer"
+
+subscriptionId=""
+resourceGroup=""
+accountName=""
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/scripts/vi_extension_install.sh:116]: Is it still needed? Did you contact arc support/office hours to figure it out?

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+param vmSubnetAddressPrefix string = '10.0.1.0/24'
+
+@description('Specifies the name of the network security group associated to the subnet hosting the virtual machine.')
+param vmSubnetNsgName string = 'VmSubnetNsg'
+
+var vnetName = 'Vm${prefix}vnet'
+var subnetName = 'Vm${vnetName}-subnet0'
+var nicName = 'Vm${prefix}nic'
+
+var vmName = 'k8s-master'
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:23]: use the prefix in the vm's name

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+@description('Specifies the address prefixes of the virtual network.')
+param virtualNetworkAddressPrefixes string = '10.0.0.0/16'
+
+@description('Specifies the address prefix of the subnet which contains the virtual machine.')
+param vmSubnetAddressPrefix string = '10.0.1.0/24'
+
+@description('Specifies the name of the network security group associated to the subnet hosting the virtual machine.')
+param vmSubnetNsgName string = 'VmSubnetNsg'
+
+var vnetName = 'Vm${prefix}vnet'
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:19]: shouldn't the prefix be in the beginning of the names?

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep [hunk 9]
```diff
@@ -0,0 +1,215 @@
+@description('Azure Resource Name Prefux')
+param prefix string
+
+@description('Specifies control plane node VM size.')
+param controlPlaneNodeVmSize string = 'Standard_DS32a_v4'
```
Comment (yehiyam) [inline VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:5]: is one DS32 enough to index? did you try to upload a movie?

File: VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh (hunk 5 above)
Comment (tshaiman) [reply VideoIndexerEnabledByArc/scripts/install_aks_cluster.sh:161]: moved to using the latest from main branch

File: VideoIndexerEnabledByArc/scripts/vi_extension_install.sh (hunk 6 above)
Comment (tshaiman) [reply VideoIndexerEnabledByArc/scripts/vi_extension_install.sh:116]: not yet

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep (hunk 7 above)
Comment (tshaiman) [reply VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:23]: this is by-design . as I had issues with the local-path provisioner which did not work on any node name with ENV Variable 
see local-path-provisioner.yaml line 131

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep (hunk 8 above)
Comment (tshaiman) [reply VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:19]: copy paste from OSS ;-)

File: VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep (hunk 9 above)
Comment (tshaiman) [reply VideoIndexerEnabledByArc/single-node-k8s-cluster/single-node.k8s.bicep:5]: not yes, changed to D64.

**Pull request closed without merging**